### PR TITLE
Logger

### PR DESCRIPTION
**Pull request by SimonHashtag**
- Finished csv-logger
- Added filters for smoothing and production of aggregate statistics
- Added overarching logger LifecycleLogger that allows simple inclusion of other loggers (tensorboardX e.g.)

Todo:
- Add median to aggregate statistics
- Add other loggers
- Make aggregate statistics filter independent of feature "Training steps" or alternatively add check for condition

### CODE REVIEW

**Comment by SimonHashtag** [comment]
**Logger that can log progress of experiments to csv-files and to tensorboard.**
- Both logging of results and hyperparameters is possible.
- Further, two filters can be applied to logs (one for smoothing and one for production of aggregate statistics like mean, std, min, max).
- Metric mean absolute fractional error (MAFE) is delivered as well (though not implemented, yet).

**Review by mesjou — APPROVED** [review]
